src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateVersions = [];
+  let stClone = { ...state };
+
+  for (const order of actions) {
+    switch (order.type) {
+      case 'addProperties':
+        stClone = { ...stClone };
```

Comment:
I don't think this reassign is necessary

Reply:
I agree with @Alireza-Mohammadhossein